app/src/main/java/org/kiwix/kiwixmobile/zim_manager/ZimFileReader.kt

```diff
@@ -90,8 +90,18 @@ class ZimFileReader(
   fun searchSuggestions(prefix: String, count: Int) =
     jniKiwixReader.searchSuggestions(prefix, count)
 
-  fun getNextSuggestion(): String? =
-    valueOfJniStringAfter(jniKiwixReader::getNextSuggestion)
+  fun getNextSuggestion(): HashMap<String, String>? {
+    val title = JNIKiwixString()
+    val url = JNIKiwixString()
+    if (jniKiwixReader.getNextSuggestion(title, url)) {
+      val results = HashMap<String, String>()
```

Comment:
This has no reason to be a hashmap, create a data class "SearchResult" and return that instead

Reply:
@Aditya-Sood ?

Reply:
@macgills does [this](https://github.com/kiwix/kiwix-android/pull/1537/commits/78cba41b81c3da82f37b9883132aac39e05db023#diff-7b697b8a92a7d0addc3f1b8deaa0767dR21) work?